pages/mercury_tour_page.py

```diff
@@ -1,73 +1,142 @@
+            EC.visibility_of_element_located((By.NAME, 'fromPort'))
+        )).select_by_visible_text(fromport)
+        return fromport
+
+    def select_departing_month(self, frommonth):
+        """
+        Select from_month from dropdown
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'fromMonth'))
+        )).select_by_visible_text(frommonth)
+        return frommonth
+
+    def select_departing_day(self, fromday):
+        """
+        Select from_day from dropdown
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'fromDay'))
+        )).select_by_visible_text(fromday)
+        return fromday
+
+    def select_arriving_port(self, toport):
+        """
+        Select to_port
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'toPort'))
+        )).select_by_visible_text(toport)
+        return toport
+
+    def select_returning_month(self, tomonth):
+        """
+        Select to_month from list
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'toMonth'))
+        )).select_by_visible_text(tomonth)
+        return tomonth
+
+    def select_returning_day(self, today):
+        """
+        Select to_day from list
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.NAME, 'toDay'))
+        )).select_by_visible_text(today)
+        return today
+
+    def select_service_class(self):
+        """
+        Select Business class from dropdown
+        """
+        WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.CSS_SELECTOR, 'input[value="Business"]'))
+        ).click()
+        service = self.driver.find_element_by_css_selector('input[value="Business"]')
+        from nose.tools import set_trace;set_trace()
+        print service
+        #return service
+
+
+    def select_an_airline(self, airline):
+        """
+        Choose an airline
+        """
+        Select(WebDriverWait(self.driver, 10).until(
+            EC.visibility_of_element_located((By.CSS_SELECTOR, 'select[name="airline"]'))
+        )).select_by_visible_text(airline)
+        return airline
+
+    def submit_flight_details_and_back_to_frontpage(self):
+        """
+        After entering all credentials submit form and wait to open next page
+        """
         self.driver.find_element_by_css_selector('input[name="findFlights"]').click()
-
-    def hotels_ver_click(self):
-        # Click hotels vertical
-        self.driver.find_element_by_link_text('Hotels').click()
-
-    def back_click(self):
-        self.driver.find_element_by_css_selector('img[src="images/home.gif"]').click()
+        WebDriverWait(self.driver, 10).until(
```

Comment:
rather than going back to main page, we should look for some heading/results on the this resulting page.

Reply:
No back click function any more